### PR TITLE
Implemented Company Prefix param

### PR DESCRIPTION
**Pull request by cloudchristoph**
# Overview/Summary

Implements the `parCompanyPrefix` from `hubNetworking`.

## This PR fixes/adds/changes/removes

1. Adds `parCompanyPrefix` as a param.

### Breaking Changes

_none_

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [ ] Associated it with relevant [ADO items](https://dev.azure.com/unifiedactiontracker/Solution%20Engineering/_backlogs/backlog/Azure%20Landing%20Zone%20Bicep/Backlog%20Bugs%20Feedback)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [ ] Performed testing and provided evidence.
- [ ] Updated relevant and associated documentation.

### CODE REVIEW

**Comment by jtracey93** [comment]
Hey @cloudchristoph,

Thanks for the PR and you efforts here. Really appreciated 👍

For this one I'm afraid we don't want to make this change in isolation and we actually are tracking a naming feature all up internally to follow the CAF naming recommendations as much as we can, where it makes sense. 

So if you are okay we will close this PR as it's not something we will do in this way. 

The reason we haven't created the names in the way you suggest is to provide the flexibility to the end user via parameter inputs so they can call the resources whatever they like. 

Whilst your approach also does this as users can override the default input. If a user does this, parCompanyPrefix will then be unused and this will mean an unrequited parameter and the linter may pick this up on build/deployment and fail the deployment. 

So in summary we want to leave it as it today as we provide the most flexibility. And we are reviewing naming all up to align to CAF naming and we will publish a naming standard we followed. 👍

Really appreciate the PR and effort here though ❤️

**Comment by cloudchristoph** [comment]
Hey @jtracey93,

thanks for your feedback. Totally fine with this.
I read out that with this also in the `hubNetworking` module this parameter will be removed in the future. Not today or tomorrow, but as soon as the naming scheme is in place, right? 

If you need any help with the modules or the schema: I'm happy to help!
